core/src/block_sync.rs

```diff
@@ -137,11 +131,21 @@ impl<S: SumeragiTrait, W: WorldTrait> Handler<ContinueSync> for BlockSynchronize
 impl<S: SumeragiTrait, W: WorldTrait> Handler<Message> for BlockSynchronizer<S, W> {
     type Result = ();
     async fn handle(&mut self, message: Message) {
-        message.handle(&mut self).await
+        message.handle(&mut self).await;
     }
 }
 
 impl<S: SumeragiTrait + Debug, W: WorldTrait> BlockSynchronizer<S, W> {
+    /// Sends request for latest blocks to a chosen peer
+    async fn request_latest_blocks_from_peer(&mut self, peer_id: PeerId) {
+        Message::GetBlocksAfter(GetBlocksAfter::new(
+            self.wsv.latest_block_hash(),
```

Comment:
If the chosen peer does not have the `latest_block_hash`,
while `GetBlocksAfter` handling it will log `ParentHashNotFound` error .
Should it be logged as an ERROR?

Reply:
I reduced log level to `warn`

Reply:
I still don't know if that's okay.
Before the PR, where receiving `LatestBlock` was followed by `GetBlocksAfter` request,
`ParentHashNotFound` did indicate some kind of fraud and ERROR was appropriate.

Now the error includes both possibilities:
1. Not found because the hash is new to the responder -- common case
2. Not found because the hash is invalid -- real error case

Reply:
As far as I can see both possibilities were there before this PR too. 
Peer would send it's latest block hash to all other peers and other peers would just check that block is not it's last block hash and not whether or not it actually exists in it's chain. If it did exist in it's chain as one of the previous blocks GetBlocksAfter would error out. I don't see any difference there

If you'd like, what I can propose is to also send height of block together with block hash in the `LatestBlock` message and in that way mitigate first error case.

Reply:
You are right.
I overlooked the possibility that the requester was a few blocks ahead.
Although it is a common case, ERROR seems to have been logged from before the PR

Reply:
so I returned the `error` log level. Are we good here?

I don't think it's possible to differentiate whether a given hash is invalid or it's ahead in the chain from the perspective of one peer